### ISSUE TITLE
@Unique: replace on conflict flag

### ISSUE THREAD

**Issue by dimitribocquet** [objectbox/objectbox-dart#297]
I am new to Objectbox, but it seems to fit to my needs, except for one thing: **upsert data**.
I fetch some items to my API server (300 items) and they have a UUID, with **@Unique**. Sometimes, these items are updated and I would like to fetch/update them, by their UUID.

FYI this is what I'm doing currently, I don't know if it's the best way:

```dart
List<CardModel> cards;

cards.forEach((card) {
        card.extension.target = extension;

        CardModel? cardInDb = _cardRepository.findByUuid(card.uuid);

        if (cardInDb != null) {
          card.id = cardInDb.id;
        }
});

_cardRepository.box.putMany(cards);
```

After some researches, I've seen that Objectbox Java has a OnConflict flag for @Unique:  

`@Unique(onConflict = ConflictStrategy.REPLACE)`
 See https://github.com/objectbox/objectbox-java/issues/509

It would be great if we have the same thing in Dart/Flutter.

**Comment by greenrobot-team** [comment]
This can now be tested with the latest preview release: `1.3.0-dev.0`

See https://docs.objectbox.io/entity-annotations#unique-constraints for an example.

**Comment by greenrobot-team** [comment]
This is now available as part of the `1.3.0` release.
